Automattic/wp-calypso
# composite-checkout: Support adding premium themes via URL

#### Changes proposed in this Pull Request

This PR adds support for adding premium themes via the special URL `/checkout/example.com/theme:themename`.

To improve the reliability of this process in the future, it also moves the `createItemToAddToCart` function to a Typescript file and cleans up quite a few incorrect or missing types in `lib/cart-items`.

#### Testing instructions

- Visit a URL like this: http://calypso.localhost:3000/checkout/example.com/theme:ovation except with your site instead of `example.com`.
- Verify that you see composite checkout and not old checkout. If you see old checkout it may be because your user is not in the abtest; you can change that via the dev menu in the lower right. If you change the abtest setting, be sure to empty your cart and try the above URL again.
- Verify that the "Ovation" premium theme has been successfully added to your cart.
